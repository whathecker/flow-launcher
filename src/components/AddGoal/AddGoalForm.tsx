/* eslint-disable no-console */
/* eslint-disable @typescript-eslint/no-unused-vars */
import React from "react";
import { Keyboard, TextInput, StyleSheet } from "react-native";
import { View, TouchableWithoutFeedback } from "../Themed";
import {
  AddGoalFormLabel,
  AddGoalRadioOption,
  AddGoalErrMsg,
} from "./components";
import { Button } from "../shared";

import { Container, Typography } from "../../styles";
import { Formik } from "formik";
import * as Yup from "yup";

type AddGoalFormProps = {
  submitHandler: () => void;
};

interface AddGoalFormValues {
  title: string;
  motivation: string;
  reminder: string;
}

const ValidationScheme = Yup.object().shape({
  title: Yup.string().required("Required"),
  motivation: Yup.string().required("Required"),
  reminder: Yup.string().required("Please select one of the options"),
});

const AddGoalForm: React.FC<AddGoalFormProps> = ({
  submitHandler,
}: AddGoalFormProps) => {
  return (
    <Formik
      initialValues={
        { title: "", motivation: "", reminder: "" } as AddGoalFormValues
      }
      validationSchema={ValidationScheme}
      onSubmit={(values) => console.log(values)}
    >
      {({
        handleChange,
        setFieldValue,
        handleSubmit,
        values,
        errors,
        touched,
      }) => (
        <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
          <View>
            <View style={styles.goalTitleInputWrapper}>
              <TextInput
                style={styles.goalTitleInput}
                multiline
                onChangeText={handleChange("title")}
                placeholder={`Write your goal in a sentence`}
                placeholderTextColor={"#848484"}
                value={values.title}
              />
              {touched.title && errors.title && (
                <AddGoalErrMsg msg={errors.title} />
              )}
            </View>

            <View style={styles.movitationInputWrapper}>
              <AddGoalFormLabel text="Movivation" type="motivation" />
              <TextInput
                style={styles.motivationInput}
                multiline
                onChangeText={handleChange("motivation")}
                placeholder={`Write down why you want to achieve this goal`}
                placeholderTextColor={"#848484"}
                value={values.motivation}
              />
              {touched.motivation && errors.motivation && (
                <AddGoalErrMsg msg={errors.motivation} />
              )}
            </View>
            <View style={styles.reminderInputWrapper}>
              <AddGoalFormLabel text="Reminder" type="reminder" />
              <View
                style={{
                  paddingTop: 30,
                  width: "80%",
                  marginLeft: "5%",
                  marginRight: "10%",
                }}
              >
                <AddGoalRadioOption
                  displayText="Every Day"
                  name="daily"
                  activeValue={values.reminder}
                  pressHandler={() => {
                    setFieldValue("reminder", "daily");
                  }}
                />
                <AddGoalRadioOption
                  displayText="Every 3 Days"
                  name="three_days"
                  activeValue={values.reminder}
                  pressHandler={() => {
                    setFieldValue("reminder", "three_days");
                  }}
                />
                <AddGoalRadioOption
                  displayText="Every Week"
                  name="seven_days"
                  activeValue={values.reminder}
                  pressHandler={() => {
                    setFieldValue("reminder", "seven_days");
                  }}
                />
                {touched.reminder && errors.reminder && (
                  <AddGoalErrMsg msg={errors.reminder} />
                )}
              </View>
            </View>
            <View style={styles.buttonWrapper}>
              <Button ctaTxt={"Add goal"} pressHandler={handleSubmit} />
            </View>
          </View>
        </TouchableWithoutFeedback>
      )}
    </Formik>
  );
};

const styles = StyleSheet.create({
  goalTitleInputWrapper: {
    paddingTop: 70,
    paddingLeft: 60,
    paddingBottom: 70,
  },
  goalTitleInput: {
    ...Typography.h1,
  },
  movitationInputWrapper: {
    paddingLeft: 60,
    paddingBottom: 50,
  },
  motivationInput: {
    ...Typography.p,
    paddingTop: 2,
    paddingBottom: 20,
  },
  reminderInputWrapper: {
    paddingLeft: 60,
    paddingBottom: 30,
  },
  buttonWrapper: {
    ...Container.centerAligned,
    flex: 1,
    width: "95%",
    margin: "2.5%",
  },
});

export default AddGoalForm;
