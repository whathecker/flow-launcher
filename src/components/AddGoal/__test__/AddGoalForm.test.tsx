import * as React from "react";
import renderer from "react-test-renderer";
import AddGoalForm from "../AddGoalForm";

describe("Test AddGoalForm component", () => {
  it("Should renders AddGoalForm correctly", () => {
    const tree = renderer.create(<AddGoalForm />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
