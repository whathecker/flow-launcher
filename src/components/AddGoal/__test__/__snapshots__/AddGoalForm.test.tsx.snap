// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddGoalForm component Should renders AddGoalForm correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "height": "40%",
          "paddingBottom": 70,
          "paddingLeft": 60,
          "paddingTop": 70,
          "width": "90%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingBottom": 10,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#655C5C",
            },
            Object {
              "fontFamily": "EB-Garamond",
              "fontSize": 20,
              "paddingRight": 12,
            },
          ]
        }
      >
        Goal
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="What do you want to achieve?"
      placeholderTextColor="#848484"
      rejectResponderTermination={true}
      style={
        Object {
          "fontFamily": "EB-Garamond",
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "height": "40%",
          "paddingBottom": 50,
          "paddingLeft": 60,
          "width": "90%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingBottom": 10,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#655C5C",
            },
            Object {
              "fontFamily": "EB-Garamond",
              "fontSize": 20,
              "paddingRight": 12,
            },
          ]
        }
      >
        Movivation
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="Write down why you want to achieve this goal"
      placeholderTextColor="#848484"
      rejectResponderTermination={true}
      style={
        Object {
          "fontFamily": "Forum-Regular",
          "fontSize": 20,
          "paddingBottom": 10,
          "paddingTop": 2,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": "10%",
          "marginRight": "10%",
          "width": "80%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0FA858",
          "borderRadius": 10,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 20,
          "paddingTop": 20,
          "shadowColor": "#171717",
          "shadowOffset": Object {
            "height": 5,
            "width": 3,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 3,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
            },
            Object {
              "fontFamily": "Forum-Regular",
              "fontSize": 20,
            },
          ]
        }
      >
        Add goal
      </Text>
    </View>
  </View>
</View>
`;
